yb-voyager/src/utils/s3/s3utils.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright (c) YugaByte, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package s3
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
+
+	"gocloud.dev/blob/s3blob"
+)
+
+var client *s3.Client
+
+func createClientIfNotExists() {
+	if client != nil {
+		return
+	}
+	cfg, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		utils.ErrExit("load s3 config: %v", err)
+	}
+	client = s3.NewFromConfig(cfg)
+}
+
+func VerifyS3FromDataDir(datadir string) error {
```

Comment:
The name of the function is not matching its implementation.

Reply:
We are simply verifying that the given s3 URL is not a malformed one, is there something else I should be naming this function?

Reply:
Your above response itself gives the name: `validateObjectURL(...)`